### ISSUE TITLE
TypeError: type object argument after ** must be a mapping, not Sphinx

### ISSUE THREAD

**Issue by ThomasDevoogdt** [sphinx-doc/sphinx#4878]
Subject: TypeError: type object argument after ** must be a mapping, not Sphinx

### Problem
- Cant use sphinx because the sphinxcontrib.googleanalytics package throws an error.

#### Procedure to reproduce the problem
```
make html
sphinx-build -b html -d build/doctrees   source build/html
```
##### Python 3.6.3
```
$ sphinx-build -b html -d build/doctrees   source build/html
Running Sphinx v1.7.2
WARNING: extension 'sphinxcontrib.googleanalytics' returned an unsupported object from its setup() function; it should return None or a metadata dictionary

Exception occurred:
  File "/usr/local/lib/python3.6/dist-packages/sphinx/registry.py", line 344, in load_extension
    app.extensions[extname] = Extension(extname, mod, **metadata)
TypeError: type object argument after ** must be a mapping, not Sphinx
The full traceback has been saved in /tmp/sphinx-err-c4n8npty.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
Makefile:55: recept voor doel 'html' is mislukt
make: *** [html] Fout 2
```

```
# Sphinx version: 1.7.2
# Python version: 3.6.3 (CPython)
# Docutils version: 0.14 
# Jinja2 version: 2.10
# Last messages:

# Loaded extensions:
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/sphinx/cmdline.py", line 303, in main
    args.warningiserror, args.tags, args.verbosity, args.jobs)
  File "/usr/local/lib/python3.6/dist-packages/sphinx/application.py", line 191, in __init__
    self.setup_extension(extension)
  File "/usr/local/lib/python3.6/dist-packages/sphinx/application.py", line 411, in setup_extension
    self.registry.load_extension(self, extname)
  File "/usr/local/lib/python3.6/dist-packages/sphinx/registry.py", line 344, in load_extension
    app.extensions[extname] = Extension(extname, mod, **metadata)
TypeError: type object argument after ** must be a mapping, not Sphinx
```

##### Python 2.7.12
```
$ python2.7 /usr/local/bin/sphinx-build -b latex -d build/doctrees   source build/latex
Running Sphinx v1.7.2
WARNING: extension 'sphinxcontrib.googleanalytics' returned an unsupported object from its setup() function; it should return None or a metadata dictionary

Exception occurred:
  File "/usr/local/lib/python2.7/dist-packages/sphinx/registry.py", line 344, in load_extension
    app.extensions[extname] = Extension(extname, mod, **metadata)
TypeError: type object argument after ** must be a mapping, not Sphinx
The full traceback has been saved in /tmp/sphinx-err-AqY4hJ.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
```

```
# Sphinx version: 1.7.2
# Python version: 2.7.12 (CPython)
# Docutils version: 0.14 
# Jinja2 version: 2.10
# Last messages:

# Loaded extensions:
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/sphinx/cmdline.py", line 303, in main
    args.warningiserror, args.tags, args.verbosity, args.jobs)
  File "/usr/local/lib/python2.7/dist-packages/sphinx/application.py", line 191, in __init__
    self.setup_extension(extension)
  File "/usr/local/lib/python2.7/dist-packages/sphinx/application.py", line 411, in setup_extension
    self.registry.load_extension(self, extname)
  File "/usr/local/lib/python2.7/dist-packages/sphinx/registry.py", line 344, in load_extension
    app.extensions[extname] = Extension(extname, mod, **metadata)
TypeError: type object argument after ** must be a mapping, not Sphinx
```

### Reproducible project / your project
- https://github.com/nickeubank/data-analysis-in-python

### Environment info
- OS: Kernel: x86_64 Linux 4.10.0-22-generic
- Python version: Python 3.6.3 & 2.7.12
- Sphinx version: sphinx-build 1.7.2

**Comment by jfbu** [comment]
Yes, the sphinxcontrib-googleanalytics installed by pip contains

```
def setup(app):
    app.add_config_value('googleanalytics_id', '', 'html')
    app.add_config_value('googleanalytics_enabled', True, 'html')
    app.connect('html-page-context', add_ga_javascript)
    app.connect('builder-inited', check_config)
    return app
```

but the source available [here](https://github.com/sphinx-contrib/googleanalytics/blob/master/sphinxcontrib/googleanalytics.py) contains

```
def setup(app):
    app.add_config_value('googleanalytics_id', '', 'html')
    app.add_config_value('googleanalytics_enabled', True, 'html')
    app.connect('html-page-context', add_ga_javascript)
    app.connect('builder-inited', check_config)
return {'version': '0.1'}
```

Only the latter version is compatible with Sphinx loading of extension. But the metada of the pip installed one does say version is ``0.1``. And [bitbucket source](https://bitbucket.org/birkenfeld/sphinx-contrib/src/8e295053a27d5914127fbb01e2a8fb57421897a7/googleanalytics/sphinxcontrib/googleanalytics.py?at=default&fileviewer=file-view-default) has the correct return statement.

It seems [commit from 2015-08-11](https://bitbucket.org/birkenfeld/sphinx-contrib/commits/cb743bfc1c9cf1258336e5ebacc915ab9a01a1aa) is not merged into pip distributed package???

Indeed from [there](https://pypi.org/project/sphinxcontrib-googleanalytics/#history) it seems afaict that the pip packaging was not updated. Can you report it upstream to the bitbucket site as an issue?

**Comment by ThomasDevoogdt** [comment]
I have [reported](https://bitbucket.org/birkenfeld/sphinx-contrib/issues/199/) the issue to their bitbucket repo. For now, I manually changed that rule of code. 

Thanks!

**Comment by tk0miya** [comment]
Indeed, this problem came from wrong extension. But Sphinx should not be crashed even if broken extensions are loaded.
So I'll work for it from now on.

**Comment by tk0miya** [comment]
Fixed by #4883.
Thank you for reporting!
